### ISSUE TITLE
Library not loaded: @rpath/libIrrXMLd.dylib

### ISSUE THREAD

**Issue by nhuyvan** [assimp/assimp#2872]
When I ran a C++ application in XCode on MacOS 10.14 that uses OpenGL and Assimp, everything compiled, but I got this error

```
dyld: Library not loaded: @rpath/libIrrXMLd.dylib
  Referenced from: /usr/local/lib/libassimpd.5.dylib
  Reason: image not found
Program ended with exit code: 9
```

Searched on Google and seems like no one has had this problem before

**Comment by kimkulling** [comment]
Could you please show the result of a ls for the folder /usr/local/lib/?
The assimp-dll depends on the libIrrXMLd.so lib. When the installation folder is not correct of there is an issue with the name for the so you will get this issue.

Another question from my side: is it correct that you have installed the debug version of assimp on your system? Or have you used a package manager to install assimp on your machine? 

Thanks in advance!

**Comment by nhuyvan** [comment]
Thank you @kimkulling for the reply. Below is the content of `/usr/local/lib`, if you are looking for `libIrrXMLd.dylib`, it's not there. I built  my own dynamic library for Assimp, I cloned the master branch from this repo, then used CMake and XCode to build the library (I used the CMake default options)

```
aspell-0.60
bash
cmake
docker
gio
glib-2.0
httpd
libGLEW.2.1.0.dylib
libGLEW.2.1.dylib
libGLEW.a
libGLEW.dylib
libargon2.1.dylib
libargon2.a
libargon2.dylib
libaspell.15.dylib
libaspell.dylib
libassimpd.5.0.0.dylib
libassimpd.5.dylib
libassimpd.dylib
libbrotlicommon-static.a
libbrotlicommon.1.0.7.dylib
libbrotlicommon.1.dylib
libbrotlicommon.dylib
libbrotlidec-static.a
libbrotlidec.1.0.7.dylib
libbrotlidec.1.dylib
libbrotlidec.dylib
libbrotlienc-static.a
libbrotlienc.1.0.7.dylib
libbrotlienc.1.dylib
libbrotlienc.dylib
libcares.2.dylib
libcares.a
libcares.dylib
libct.4.dylib
libct.a
libct.dylib
libecomlodr.dylib
libev.4.dylib
libev.a
libev.dylib
libevent-2.1.7.dylib
libevent.a
libevent.dylib
libevent_core-2.1.7.dylib
libevent_core.a
libevent_core.dylib
libevent_extra-2.1.7.dylib
libevent_extra.a
libevent_extra.dylib
libevent_openssl-2.1.7.dylib
libevent_openssl.a
libevent_openssl.dylib
libevent_pthreads-2.1.7.dylib
libevent_pthreads.a
libevent_pthreads.dylib
libfreetype.6.dylib
libfreetype.a
libfreetype.dylib
libgdbm.6.dylib
libgdbm.a
libgdbm.dylib
libgio-2.0.0.dylib
libgio-2.0.dylib
libglfw.3.3.dylib
libglfw.3.dylib
libglfw.dylib
libglib-2.0.0.dylib
libglib-2.0.dylib
libgmodule-2.0.0.dylib
libgmodule-2.0.dylib
libgmp.10.dylib
libgmp.a
libgmp.dylib
libgmpxx.4.dylib
libgmpxx.a
libgmpxx.dylib
libgobject-2.0.0.dylib
libgobject-2.0.dylib
libgthread-2.0.0.dylib
libgthread-2.0.dylib
libidn.12.dylib
libidn.a
libidn.dylib
libjansson.4.dylib
libjansson.a
libjansson.dylib
libjemalloc.2.dylib
libjemalloc.a
libjemalloc.dylib
libjemalloc_pic.a
libjpeg.9.dylib
libjpeg.a
libjpeg.dylib
libltdl.7.dylib
libltdl.a
libltdl.dylib
liblzma.5.dylib
liblzma.a
liblzma.dylib
libmetalink.3.dylib
libmetalink.a
libmetalink.dylib
libmsodbcsql.17.dylib
libnghttp2.14.dylib
libnghttp2.a
libnghttp2.dylib
libodbc.2.dylib
libodbc.a
libodbc.dylib
libodbccr.2.dylib
libodbccr.a
libodbccr.dylib
libodbcinst.2.dylib
libodbcinst.a
libodbcinst.dylib
libpcre.1.dylib
libpcre.a
libpcre.dylib
libpcre16.0.dylib
libpcre16.a
libpcre16.dylib
libpcre2-16.0.dylib
libpcre2-16.a
libpcre2-16.dylib
libpcre2-32.0.dylib
libpcre2-32.a
libpcre2-32.dylib
libpcre2-8.0.dylib
libpcre2-8.a
libpcre2-8.dylib
libpcre2-posix.2.dylib
libpcre2-posix.a
libpcre2-posix.dylib
libpcre32.0.dylib
libpcre32.a
libpcre32.dylib
libpcrecpp.0.dylib
libpcrecpp.a
libpcrecpp.dylib
libpcreposix.0.dylib
libpcreposix.a
libpcreposix.dylib
libpng.a
libpng.dylib
libpng16.16.dylib
libpng16.a
libpng16.dylib
libpspell.15.dylib
libpspell.dylib
librtmp.1.dylib
librtmp.a
librtmp.dylib
libsodium.23.dylib
libsodium.a
libsodium.dylib
libssh2.1.dylib
libssh2.a
libssh2.dylib
libsybdb.5.dylib
libsybdb.a
libsybdb.dylib
libtdsodbc.0.so
libtdsodbc.a
libtdsodbc.so
libtidy.5.6.0.dylib
libtidy.5.dylib
libtidy.dylib
libtidys.a
libtiff.5.dylib
libtiff.a
libtiff.dylib
libtiffxx.5.dylib
libtiffxx.a
libtiffxx.dylib
libwebp.7.dylib
libwebp.a
libwebp.dylib
libwebpdecoder.3.dylib
libwebpdecoder.a
libwebpdecoder.dylib
libwebpdemux.2.dylib
libwebpdemux.a
libwebpdemux.dylib
libwebpmux.3.dylib
libwebpmux.a
libwebpmux.dylib
libzip.5.0.dylib
libzip.5.dylib
libzip.dylib
output.txt
php
pkgconfig
python3.7
```

**Comment by kimkulling** [comment]
So it seems that the assimp-configuration thought that you want to use the system-based irrXml library instead of using the version assimp deploys. 
Do you have used the following option when building assimp:

`cmake -DSYSTEM_IRRXML=ON ...`

Or could you please check the CMakeCache content of your build for the variable 

`SYSTEM_IRRXML`

It must be set to 

'-DSYSTEM_IRRXML=OFF'

Which shall be the default. But we make a lot of tryouts for MacOS because our buildchain does not include this platform at the moment. So sorry for the inconeviances!

**Comment by nhuyvan** [comment]
I just left everything in their default options, I am very positive that when I compiled Assimp, `SYSTEM_IRRXML` was not checked, so it had value OFF. I just retried compiling Assimp, it is now working for me, no morrow image not loaded error. When I launched CMake, in its GUI, it showed a list of options, and there was this option that said something about DEBUG (I coudn't remember exactly what it was), and I accidentally deleted that option, then the error was gone. I am not sure why, but in your very comment to my issue, you mentioned something about not using the debug version, I think me deleting that SOMETHING_DEBUG_SOMETHING option fixed my problem. Thank you very much @kimkulling
